de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java

```diff
@@ -657,7 +657,16 @@ public void run() {
 		 */
 		public void finish() {
 			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "ResourceChangeEventThread: Thread triggered for stop.... "));
-			threadFinished = true;
+			triggerFinished = true;
+			
+			// Now wait until the thread actually terminated
+			while (this.isAlive()) {

```

Comment:
This looks like a re-implementation of the join() method of the Thread class.

Reply:
Right. Done!